src/server/db/database.router.js

```diff
@@ -28,6 +29,27 @@ class DatabaseRouter {
       });
   }
 
+  async getComponentDashboardInfoRequest(req: express.$Request, res: express.$Response) {
+    const componentName = req.query.component;
+    if (!req.user) {
+      res.sendStatus(401);
+      return;
+    }
+
+    const username = req.user.username;
+    const dashboardInfo = await this.database.getComponentDashboardInfo(username, componentName);
+    const deploying = await this.database.getComponentDeployingStatus(username, componentName);
+    const result = {
+      deploymentIntention: dashboardInfo.deployment_intention,
+      receivedTime: dashboardInfo.received_time,
+      color: dashboardInfo.color,
+      isDeploying: deploying != null,
+      deploymentReason: deploying != null ? deploying.deployment_reason : ''
```

Comment:
I don't like returning the empty string as a sentinel. Is there a way to avoid this?